charts/victoria-metrics-distributed/README.md

```diff
@@ -0,0 +1,283 @@
+The default values suppose there are two availability zones with label `topology.kubernetes.io/zone: zone-a` and `topology.kubernetes.io/zone: zone-a`, default topology will be like below:
+![victoriametrics-distributed-topology](./victoriametrics-distributed-topology.jpg)
+
+It contains required availability zone components and global components, and other optional resources.
+
+Availability zone components:
+1. vmagent: remotes write to all availability zones to ensure data completeness.
+2. vmauth-ingest: performs `least_loaded` policy for spreading requests to vminsert instances located in the same availability zone.
+3. vmcluster: supports different setup per availability zone.
+4. vmauth-query: performs `least_loaded` policy for spreading requests to vmselect instances located in the same availability zone.
+5. vmauth-query-cross-az: performs `first_available` policy on vmselect instances of all the availability zones, prefers "local" vmselect to reduce cross zone network traffic.
+
+Global components:
+1. vmagent(optional): an extra vmagent to scrape targets and remote write to `vmauth-global-ingest`.
+2. vmauth-global-ingest: global write entrance.
+3. vmauth-global-query: global query entrance.
+4. grafana(optional): uses `vmauth-global-query` as default datasource.
+
+Note:
+As the diagram showed above, this chart doesn't include components like vmalert, alertmanager etc by default. If needed, you can create those resources using dependency `victoria-metrics-k8s-stack` chart.
+
+### When&Why use `victoria-metrics-distributed` chart?
+
+One of the best practice of running production kubernetes cluster is running with [multiple availability zones](https://kubernetes.io/docs/setup/best-practices/multiple-zones/). And apart from kubernetes control plane components, we also want to spread our application pods on multiple zones to continue working during zone outage.
+There are two ways to set up VictoriaMetrics cluster on multiple availability zones:
+1. using one vmcluster but spread the components on different zones by `topologySpreadConstraints`.
+For example, set up vmcluster with `replicas: 2` for vminsert, vmstorage and vmselect with `replicationFactor=2`, and set `topologySpreadConstraints` for all of the components:
+```
+    topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+```
+Each zone will have a pair of nodes of each type - vminsert, vmselect and vmstorage, thus vmcluster can still serve both write and query when one zone is compeletely down.
+But VictoriaMetrics components themselves don't support zone-awareness, means in the above example, if you have three vmstorage nodes with `replicationFactor=2`, it can't be ensure that duplicated data are written to vmstorage instances on different availability zone, thus data could be incomplete when one availability zone is down.
```

Comment:
Doesn't this mean this approach is incorrect?) If yes, we shouldn't mention it.

Reply:
yes, removed